### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by OlivierLi**
- Now support local chromium build.
- Add a new --profile execution mode. (Runs the wake-ups probe for now, normal stack collection will be added after.)

### CODE REVIEW

File: benchmark.py [hunk 1]
```diff
@@ -39,28 +41,41 @@ def RunScenario(scenario_config):
   if scenario_config.background_script is not None:
     subprocess.call(["osascript", f'./driver_scripts/{background_script}.scpt'])
 
-  subprocess.call(["osascript", f'./driver_scripts/{scenario_config.driver_script}.scpt'])
+  driver_script_args = ["osascript", f'./driver_scripts/{scenario_config.driver_script}.scpt']
+
+  process = subprocess.Popen(driver_script_args)
+  time.sleep(2)
```
Comment (eti-p-doray) [inline benchmark.py]: Comment why sleep?

File: benchmark.py [hunk 2]
```diff
@@ -39,28 +41,41 @@ def RunScenario(scenario_config):
...
+
+def Profile(scenario_config, output_dir, dry_run=False):
+  script_process = RunScenario(scenario_config)
 
   if scenario_config.browser != "Chromium":
     print("Only Chromium can be profiled!")
     exit(-1)
 
-  browser_executable = utils.browsers_definition[scenario_config.browser]['executable']
+  browser_executable = utils.browsers_definition[scenario_config.browser]['process_name']
```
Comment (eti-p-doray) [inline benchmark.py]: Nit: browser_process = ...

File: benchmark.py [hunk 3]
```diff
@@ -29,7 +31,7 @@ def RunScenario(scenario_config):
 
   if scenario_config.browser is not None:
     browser_executable = utils.browsers_definition[scenario_config.browser]['executable']
-    if scenario_config.browser in ["Chrome", "Canary", "Edge"]:
+    if scenario_config.browser in ["Chromium", "Chrome", "Canary", "Edge"]:
```
Comment (eti-p-doray) [inline benchmark.py:34]: I don't think we want "open -a ..." for Chromium, since it's a full path, but rather "open ..."

File: utils.py [hunk 4]
```diff
@@ -1,24 +1,30 @@
...
     "identifier": "com.google.Chrome"
   },
   "Canary": {
     "executable": "Google Chrome Canary",
+    "process_name": "Google Chrome Canary",
     "identifier": "com.google.Chrome.canary"
   },
   "Chromium": {
-    "executable": "Chromium",
+    "executable": f"/Users/{user_name}/git/chromium/src/out/Release/Chromium.app/Contents/MacOS/Chromium",
```
Comment (eti-p-doray) [inline utils.py]: The part "Users/{user_name}/git/chromium/src/out/Release" may vary (especially with multiple builds). Do you think we can make this an argument of ScenarioConfig or something?

Comment (OlivierLi) [reply utils.py]: Good point, I made it enforced in ScenarioConfig but passed in through a command line argument.

File: benchmark.py (hunk 1 above)
Comment (OlivierLi) [reply benchmark.py]: Good point, made this much cleaner and minimized the length of the sleep.

File: benchmark.py (hunk 2 above)
Comment (OlivierLi) [reply benchmark.py]: Changed.

File: benchmark.py (hunk 3 above)
Comment (OlivierLi) [reply benchmark.py:34]: Humm, it seems to work fine as is. Do you know if it does anything bad to leave as is?

Comment (eti-p-doray) [reply benchmark.py:34]: Heh, if it works that's fine :)

**Review by eti-p-doray — APPROVED** [review]

**Pull request merged**